### PR TITLE
Change the varible names that keep track of rr policy its state and

### PR DESCRIPTION
**Pull request by vishalpowar**
resolution tracking to remove reference of the policy type (RR)

Not toucing the function names as these functions represent what they
are doing (Creating or working with RR policy).

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/17163)

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  +0.0%     +64 [None]     +16  +0.0%

  +0.0%     +64 TOTAL      +16  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,149      Total (>)      2,019,040

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,736      Total (<)     11,179,738

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by vishalpowar** [comment]
Right now there is no parent_policy in xDSlb. So naming it child_policy does not really add anything here.
Also, we are going to wrap this into LocalityWrapper, at that time we will have to rename it again. So I would prefer to keep this as policy

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 1]
```diff
@@ -273,7 +273,7 @@ class XdsLb : public LoadBalancingPolicy {
   bool PickFromRoundRobinPolicyLocked(bool force_async, PendingPick* pp,
                                       grpc_error** error);
   void UpdateConnectivityStateFromRoundRobinPolicyLocked(
-      grpc_error* rr_state_error);
+      grpc_error* state_error);
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Suggest changing all variables that previous included the string `rr_` to instead use `child_`.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 2]
```diff
@@ -273,7 +273,7 @@ class XdsLb : public LoadBalancingPolicy {
   bool PickFromRoundRobinPolicyLocked(bool force_async, PendingPick* pp,
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Please rename all methods that include the string `RoundRobin` to instead use `Child`.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc (hunk 1 above)
Comment (vishalpowar) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Juanli had the same comment, and i could not see a reason why.
With both of you pointing out the same naming change, has convinced me to make the change. So I will.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,161      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,705      Total (<)     11,179,731

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,161      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,712      Total (<)     11,179,737

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 3]
```diff
@@ -36,20 +36,18 @@
...
-/// Once an RR policy instance is in place (and getting updated as described),
-/// calls for a pick, a ping, or a cancellation will be serviced right
-/// away by forwarding them to the RR instance.  Any time there's no RR
-/// policy available (i.e., right after the creation of the gRPCLB policy),
-/// pick and ping requests are added to a list of pending picks and pings
-/// to be flushed and serviced when the RR policy instance becomes available.
+/// Once a child policy instance is in place (and getting updated as
+/// described), calls for a pick, a ping, or a cancellation will be serviced
+/// right away by forwarding them to the policy instance.  Any time there's no
+/// child policy available (i.e., right after the creation of the gRPCLB
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/gRPCLB/xds/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 4]
```diff
@@ -36,20 +36,18 @@
...
+/// addresses.
 ///
-/// Once an RR policy instance is in place (and getting updated as described),
-/// calls for a pick, a ping, or a cancellation will be serviced right
-/// away by forwarding them to the RR instance.  Any time there's no RR
-/// policy available (i.e., right after the creation of the gRPCLB policy),
-/// pick and ping requests are added to a list of pending picks and pings
-/// to be flushed and serviced when the RR policy instance becomes available.
+/// Once a child policy instance is in place (and getting updated as
+/// described), calls for a pick, a ping, or a cancellation will be serviced
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: You can remove references to "ping" requests here.  That was removed from the LB policy API in #16167.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 5]
```diff
@@ -36,20 +36,18 @@
...
+/// right away by forwarding them to the policy instance.  Any time there's no
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 6]
```diff
@@ -141,10 +139,10 @@ class XdsLb : public LoadBalancingPolicy {
 
  private:
   /// Linked list of pending pick requests. It stores all information needed to
-  /// eventually call (Round Robin's) pick() on them. They mainly stay pending
-  /// waiting for the RR policy to be created.
+  /// eventually call pick() on them. They mainly stay pending waiting for the
+  /// policy to be created.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 7]
```diff
@@ -444,7 +442,7 @@ grpc_lb_addresses* ProcessServerlist(const xds_grpclb_serverlist* serverlist) {
   grpc_lb_addresses* lb_addresses =
       grpc_lb_addresses_create(num_valid, &lb_token_vtable);
   /* second pass: actually populate the addresses and LB tokens (aka user data
-   * to the outside world) to be read by the RR policy during its creation.
+   * to the outside world) to be read by the policy during its creation.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 8]
```diff
@@ -1134,11 +1132,12 @@ void XdsLb::HandOffPendingPicksLocked(LoadBalancingPolicy* new_policy) {
 // Cancel a specific pending pick.
 //
 // A grpclb pick progresses as follows:
-// - If there's a Round Robin policy (rr_policy_) available, it'll be
-//   handed over to the RR policy (in CreateRoundRobinPolicyLocked()). From
-//   that point onwards, it'll be RR's responsibility. For cancellations, that
-//   implies the pick needs also be cancelled by the RR instance.
-// - Otherwise, without an RR instance, picks stay pending at this policy's
+// - If there's a policy (child_policy_) available, it'll be handed over to the
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy (child_policy_)/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 9]
```diff
@@ -1134,11 +1132,12 @@ void XdsLb::HandOffPendingPicksLocked(LoadBalancingPolicy* new_policy) {
...
+// policy
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Nit: Please fix line wrapping.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 10]
```diff
@@ -1134,11 +1132,12 @@ void XdsLb::HandOffPendingPicksLocked(LoadBalancingPolicy* new_policy) {
...
+//   (in CreateChildPolicyLocked()). From that point onwards, it'll be the
+//   policy's responsibility. For cancellations, that implies the pick needs to
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 11]
```diff
@@ -1134,11 +1132,12 @@ void XdsLb::HandOffPendingPicksLocked(LoadBalancingPolicy* new_policy) {
...
+//   be also cancelled by the policy instance.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 12]
```diff
@@ -1134,11 +1132,12 @@ void XdsLb::HandOffPendingPicksLocked(LoadBalancingPolicy* new_policy) {
...
+// - Otherwise, without an policy instance, picks stay pending at this policy's
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/an policy/a child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 13]
```diff
@@ -1159,20 +1158,21 @@ void XdsLb::CancelPickLocked(PickState* pick, grpc_error* error) {
...
 
 // Cancel all pending picks.
 //
 // A grpclb pick progresses as follows:
-// - If there's a Round Robin policy (rr_policy_) available, it'll be
-//   handed over to the RR policy (in CreateRoundRobinPolicyLocked()). From
-//   that point onwards, it'll be RR's responsibility. For cancellations, that
-//   implies the pick needs also be cancelled by the RR instance.
-// - Otherwise, without an RR instance, picks stay pending at this policy's
+// - If there's a policy (child_policy_) available, it'll be handed over to the
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Same comments as above.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 14]
```diff
@@ -1213,31 +1213,29 @@ void XdsLb::ResetBackoffLocked() {
...
     if (pick->on_complete == nullptr) {
       *error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
           "No pick result available but synchronous result required.");
       pick_done = true;
     } else {
       if (grpc_lb_xds_trace.enabled()) {
         gpr_log(GPR_INFO,
-                "[xdslb %p] No RR policy. Adding to grpclb's pending picks",
-                this);
+                "[xdslb %p] No policy. Adding to grpclb's pending picks", this);
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 15]
```diff
@@ -1445,7 +1443,7 @@ void XdsLb::OnBalancerChannelConnectivityChangedLocked(void* arg,
   XdsLb* xdslb_policy = static_cast<XdsLb*>(arg);
   if (xdslb_policy->shutting_down_) goto done;
   // Re-initialize the lb_call. This should also take care of updating the
-  // embedded RR policy. Note that the current RR policy, if any, will stay in
+  // embedded policy. Note that the current policy, if any, will stay in
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/embedded/child/
s/current policy/current child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 16]
```diff
@@ -1505,7 +1503,7 @@ void DestroyClientStats(void* arg) {
 }
 
 void XdsLb::PendingPickSetMetadataAndContext(PendingPick* pp) {
-  /* if connected_subchannel is nullptr, no pick has been made by the RR
+  /* if connected_subchannel is nullptr, no pick has been made by the
    * policy (e.g., all addresses failed to connect). There won't be any
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 17]
```diff
@@ -1558,24 +1556,24 @@ void XdsLb::AddPendingPick(PendingPick* pp) {
 }
 
 //
-// code for interacting with the RR policy
+// code for interacting with the Child policy
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/Child/child/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 18]
```diff
@@ -1558,24 +1556,24 @@ void XdsLb::AddPendingPick(PendingPick* pp) {
...
+                                      grpc_error** error) {
   // Set client_stats and user_data.
   if (lb_calld_ != nullptr && lb_calld_->client_stats() != nullptr) {
     pp->client_stats = lb_calld_->client_stats()->Ref();
   }
   GPR_ASSERT(pp->pick->user_data == nullptr);
   pp->pick->user_data = (void**)&pp->lb_token;
-  // Pick via the RR policy.
-  bool pick_done = rr_policy_->PickLocked(pp->pick, error);
+  // Pick via the Child policy.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/Child/child/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 19]
```diff
@@ -1586,57 +1584,58 @@ bool XdsLb::PickFromRoundRobinPolicyLocked(bool force_async, PendingPick* pp,
     Delete(pp);
   }
   // else, the pending pick will be registered and taken care of by the
-  // pending pick list inside the RR policy.  Eventually,
+  // pending pick list inside the policy.  Eventually,
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 20]
```diff
@@ -1586,57 +1584,58 @@ bool XdsLb::PickFromRoundRobinPolicyLocked(bool force_async, PendingPick* pp,
...
+  child_policy_->ExitIdleLocked();
+  // Send pending picks to policy.
   PendingPick* pp;
   while ((pp = pending_picks_)) {
     pending_picks_ = pp->next;
     if (grpc_lb_xds_trace.enabled()) {
       gpr_log(GPR_INFO,
-              "[xdslb %p] Pending pick about to (async) PICK from RR %p", this,
-              rr_policy_.get());
+              "[xdslb %p] Pending pick about to (async) PICK from policy %p",
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 21]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
   if (shutting_down_) return;
-  grpc_channel_args* args = CreateRoundRobinPolicyArgsLocked();
+  grpc_channel_args* args = CreateChildPolicyArgsLocked();
   GPR_ASSERT(args != nullptr);
-  if (rr_policy_ != nullptr) {
+  if (child_policy_ != nullptr) {
     if (grpc_lb_xds_trace.enabled()) {
-      gpr_log(GPR_INFO, "[xdslb %p] Updating RR policy %p", this,
-              rr_policy_.get());
+      gpr_log(GPR_INFO, "[xdslb %p] Updating the policy %p", this,
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/the/child/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 22]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
     LoadBalancingPolicy::Args lb_policy_args;
     lb_policy_args.combiner = combiner();
     lb_policy_args.client_channel_factory = client_channel_factory();
     lb_policy_args.args = args;
-    CreateRoundRobinPolicyLocked(lb_policy_args);
+    CreateChildPolicyLocked(lb_policy_args);
     if (grpc_lb_xds_trace.enabled()) {
-      gpr_log(GPR_INFO, "[xdslb %p] Created new RR policy %p", this,
-              rr_policy_.get());
+      gpr_log(GPR_INFO, "[xdslb %p] Created new policy %p", this,
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 23]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
+    xdslb_policy->Unref(DEBUG_LOCATION, "on_child_reresolution_requested");
     return;
   }
   if (grpc_lb_xds_trace.enabled()) {
-    gpr_log(
-        GPR_INFO,
-        "[xdslb %p] Re-resolution requested from the internal RR policy (%p).",
-        xdslb_policy, xdslb_policy->rr_policy_.get());
+    gpr_log(GPR_INFO,
+            "[xdslb %p] Re-resolution requested from the internal policy (%p).",
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/the internal/child/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 24]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
+            xdslb_policy, xdslb_policy->child_policy_.get());
   }
   // If we are talking to a balancer, we expect to get updated addresses form
-  // the balancer, so we can ignore the re-resolution request from the RR
-  // policy. Otherwise, handle the re-resolution request using the
-  // grpclb policy's original re-resolution closure.
+  // the balancer, so we can ignore the re-resolution request from the policy.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 25]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
+  // Otherwise, handle the re-resolution request using the grpclb policy's
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/grpclb/xds/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 26]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
+  // original re-resolution closure.
   if (xdslb_policy->lb_calld_ == nullptr ||
       !xdslb_policy->lb_calld_->seen_initial_response()) {
     xdslb_policy->TryReresolutionLocked(&grpc_lb_xds_trace, GRPC_ERROR_NONE);
   }
-  // Give back the wrapper closure to the RR policy.
-  xdslb_policy->rr_policy_->SetReresolutionClosureLocked(
-      &xdslb_policy->on_rr_request_reresolution_);
+  // Give back the wrapper closure to the policy.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 27]
```diff
@@ -1664,66 +1663,65 @@ grpc_channel_args* XdsLb::CreateRoundRobinPolicyArgsLocked() {
...
 
-void XdsLb::UpdateConnectivityStateFromRoundRobinPolicyLocked(
-    grpc_error* rr_state_error) {
+void XdsLb::UpdateConnectivityStateFromChildPolicyLocked(
+    grpc_error* child_state_error) {
   const grpc_connectivity_state curr_glb_state =
       grpc_connectivity_state_check(&state_tracker_);
   /* The new connectivity status is a function of the previous one and the new
-   * input coming from the status of the RR policy.
+   * input coming from the status of the policy.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 28]
```diff
@@ -1736,52 +1734,51 @@ void XdsLb::UpdateConnectivityStateFromRoundRobinPolicyLocked(
    *  SD || NA |  NA |  NA |  NA  |  NA  | (*)
    *  ---++----+-----+-----+------+------+
    *
-   * A [STATE] indicates that the old RR policy is kept. In those cases, STATE
+   * A [STATE] indicates that the old policy is kept. In those cases, STATE
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/policy/child policy/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 29]
```diff
@@ -1736,52 +1734,51 @@ void XdsLb::UpdateConnectivityStateFromRoundRobinPolicyLocked(
...
    * is the current state of grpclb, which is left untouched.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/grpclb/xds/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 30]
```diff
@@ -1736,52 +1734,51 @@ void XdsLb::UpdateConnectivityStateFromRoundRobinPolicyLocked(
...
    *
    *  In summary, if the new state is TRANSIENT_FAILURE or SHUTDOWN, stick to
-   *  the previous RR instance.
+   *  the previous  instance.
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/instance/child policy instance/

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 31]
```diff
@@ -1736,52 +1734,51 @@ void XdsLb::UpdateConnectivityStateFromRoundRobinPolicyLocked(
...
+      GPR_ASSERT(child_state_error == GRPC_ERROR_NONE);
   }
   if (grpc_lb_xds_trace.enabled()) {
-    gpr_log(
-        GPR_INFO,
-        "[xdslb %p] Setting grpclb's state to %s from new RR policy %p state.",
-        this, grpc_connectivity_state_name(rr_connectivity_state_),
-        rr_policy_.get());
+    gpr_log(GPR_INFO,
+            "[xdslb %p] Setting grpclb's state to %s from new policy %p state.",
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: s/grpclb/xds/
s/policy/child policy/

**Comment by AspirinSJL** [comment]
@vishalpowar I think the xDS policy itself is the parent policy? That's why we need to name the LB policies it contains as child policies. It's clearer if you look at the LB config in service config. Actually, we use the exact name "child_policy" in xDS config.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.0%     +96 [None]                                                                                  +344  +0.0%
      +0.0%     +96 [Unmapped]                                                                              +344  +0.0%
      [NEW]      +8 grpc_core::(anonymous namespace)::XdsLb::CreateChildPolicyArgsLocked()::keys_to_remo      +8  [NEW]

  +0.0%     +96 TOTAL                                                                                   +344  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.0%    +112 [None]                                                                                  +344  +0.0%
      +0.0%    +112 [Unmapped]                                                                              +344  +0.0%
      [NEW]      +8 grpc_core::(anonymous namespace)::XdsLb::CreateChildPolicyArgsLocked()::keys_to_remo      +8  [NEW]

  +0.0%    +112 TOTAL                                                                                   +344  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,209      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,705      Total (<)     11,179,731

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,223      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,706      Total (<)     11,179,736

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,235      Total (>)      2,019,028

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,709      Total (<)     11,179,733

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.0%     +96 [None]                                                                                  +336  +0.0%
      +0.0%     +96 [Unmapped]                                                                              +336  +0.0%
      [NEW]      +8 grpc_core::(anonymous namespace)::XdsLb::CreateChildPolicyArgsLocked()::keys_to_remo      +8  [NEW]

  +0.0%     +96 TOTAL                                                                                   +336  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,433      Total (>)      2,019,244

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,179,081      Total (<)     11,179,120

 No significant differences in binary sizes


```

**Pull request merged**